### PR TITLE
Firefox recognizes Server-Timing header, but only partially supports Trailer headers (DevTools only)

### PR DESCRIPTION
**Pull request by bsmth**
#### Summary

The support for HTTP Trailer header is misleading, as it's removed from the Fetch spec & not accessible there anyway / XHR. As far as I can see, the only support that exists in browsers is exclusively for `Server-Timing`, and at that, it's only used for DevTools in Network -> Timing, all other headers (as trailers) are a no-op / discarded. For this reason, I've set it to partial support for `Trailer`, and full support for Server-Timing header as trailer, which no other browser appears to do.

#### Test results and supporting details

- **Firefox** `Server-Timing` implementation bug -> https://bugzilla.mozilla.org/show_bug.cgi?id=1403051:
  - Networking aspect landed in Fx59 -> https://bugzilla.mozilla.org/show_bug.cgi?id=1413999
  - Exposed in DevTools in Fx71 -> https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/71#developer_tools
  - MDN docs issue -> https://github.com/mdn/sprints/issues/2277
  - Necko source for [handling of trailers](https://github.com/mozilla/gecko-dev/blob/2deb9bcf801f9de83d4f30c890d442072b9b6595/netwerk/protocol/http/nsHttpTransaction.cpp#L3095-L3143)

- **ChrBugs** -> https://issues.chromium.org/issues/40811358

Node repro:

```js
const http = require("http");
http
  .createServer((req, res) => {
    res.writeHead(200, {
      "Content-Type": "text/plain",
      "Transfer-Encoding": "chunked",
      // Server-Timing as header
      // "Server-Timing": "custom-metric;dur=567.123",
      Trailer: "Server-Timing",
    });
    res.write("Hello.\n");
    // Server-Timing as trailer
    // See https://nodejs.org/api/http.html#responseaddtrailersheaders
    res.addTrailers({
      "Server-Timing": "custom-metric;dur=666.123",
    });
    res.end();
  })
  .listen(3000, () => {
    console.log("Server running at http://localhost:3000");
  });
```

#### Related issues

- [ ] https://github.com/mdn/browser-compat-data/issues/24684
- [ ] https://github.com/mdn/browser-compat-data/issues/18441
- [ ] https://github.com/mdn/browser-compat-data/issues/14703
- [ ] https://github.com/mdn/content/issues/7137

### CODE REVIEW

**Comment by caugner** [comment]
FWIW I only found this indirectly related WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=178069

**Review by caugner — APPROVED** [review]
LGTM.

Compact diff grouped by file:
image

File: http/headers/Server-Timing.json [hunk 1]
```diff
@@ -39,6 +39,41 @@
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "trailer": {
+          "__compat": {
+            "description": "`Server-Timing` as HTTP trailer",
+            "support": {
+              "chrome": {
+                "version_added": false
```
Comment (caugner) [inline http/headers/Server-Timing.json:48]: Oh, we should probably add the Chromium impl_url here:

https://issues.chromium.org/issues/40811358

**Review by caugner — CHANGES_REQUESTED** [review]
One nit.

File: http/headers/Server-Timing.json (hunk 1 above)
Comment (bsmth) [reply http/headers/Server-Timing.json:48]: Good idea, thanks for the remninder

Comment (bsmth) [reply http/headers/Server-Timing.json]: added in 7314336174563b2d205c66b0ecc84f6ce4dc80be

**Review by caugner — APPROVED** [review]

**Pull request merged**